core/src/mender-client.c

```diff
@@ -838,7 +850,24 @@ mender_client_authentication_work_function(void) {
 
         /* Publish deployment status */
         if (true == success) {
+
+#ifdef CONFIG_MENDER_FULL_PARSE_ARTIFACT
+#ifdef CONFIG_MENDER_PROVIDES_DEPENDS
+            mender_key_value_list_t *new_provides = NULL;
+            if (MENDER_OK != mender_storage_get_new_provides(&new_provides)) {
+                mender_log_error("Unable to get new provides");
+                mender_client_publish_deployment_status(id, MENDER_DEPLOYMENT_STATUS_FAILURE);
+                goto RELEASE;
+            }
+            if (MENDER_OK != mender_storage_set_provides(new_provides)) {
+                mender_log_error("Unable to set provides");
+                mender_client_publish_deployment_status(id, MENDER_DEPLOYMENT_STATUS_FAILURE);
+                goto RELEASE;
+            }
```

Comment:
This block is worth a comment, it's quite confusing what's going on in here. Also, do we really want to do this right after authentication?

Reply:
This will happen after a reboot and if the deployment is a success, right?. I didn't find anywhere else to put this, and figured this was the closest I came to `ArtifactCommit` after a reboot. Do you have any other suggestions?

Reply:
I guess there's no better place until we have the update module state machine and it can go into the `Commit` state. But the comment is necessary.